tfinstall/download.go

```diff
@@ -0,0 +1,133 @@
+package tfinstall
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/hashicorp/go-getter"
+	"golang.org/x/crypto/openpgp"
+)
+
+func ensureInstallDir(installDir string) (string, error) {
+	if installDir == "" {
+		return ioutil.TempDir("", "tfexec")
+	}
+
+	if _, err := os.Stat(installDir); err != nil {
+		return "", fmt.Errorf("could not access directory %s for installing Terraform: %w", installDir, err)
+	}
+
+	return installDir, nil
+}
+
+func downloadWithVerification(ctx context.Context, tfVersion string, installDir string) (string, error) {
+	osName := runtime.GOOS
+	archName := runtime.GOARCH
+
+	// setup: ensure we have a place to put our downloaded terraform binary
+	tfDir, err := ensureInstallDir(installDir)
+	if err != nil {
+		return "", err
+	}
+
+	// setup: getter client
+
+	// TODO: actually use this header...
+	// httpHeader := make(http.Header)
+	// httpHeader.Set("User-Agent", "HashiCorp-tfinstall/"+Version)
```

Comment:
@kmoe I noticed the header doesn't actually seem to be attached anywhere, but haven't gotten a chance to look at how to set it in go-getter yet. Just an FYI if you happen to poke around in here.